lib/components/DepositForm.jsx

```diff
@@ -1,41 +1,39 @@
 import React, { useContext, useEffect, useState } from 'react'
 import { ethers } from 'ethers'
 import IERC20Abi from '@pooltogether/pooltogether-contracts/abis/IERC20Upgradeable'
-import { useAtom } from 'jotai'
 import FeatherIcon from 'feather-icons-react'
 
 import { Button } from 'lib/components/Button'
 import { RightLabelButton, TextInputGroup } from 'lib/components/TextInputGroup'
 import { displayAmountInEther } from 'lib/utils/displayAmountInEther'
-import { fetchPoolChainValues, poolChainValuesAtom } from 'lib/hooks/usePoolChainValues'
-import { userChainValuesAtom } from 'lib/hooks/useUserChainValues'
+import { useUserChainValues } from 'lib/hooks/useUserChainValues'
 import { sendTx } from 'lib/utils/sendTx'
 import { WalletContext } from 'lib/components/WalletContextProvider'
-import { poolAddressesAtom } from 'lib/hooks/usePoolAddresses'
-import { contractVersionsAtom, prizePoolTypeAtom } from 'lib/hooks/useDetermineContractVersions'
-import { errorStateAtom } from 'lib/components/PoolData'
 import { useNetwork } from 'lib/hooks/useNetwork'
 import { InnerCard } from 'lib/components/Card'
 import { numberWithCommas } from 'lib/utils/numberWithCommas'
 
 import Warning from 'assets/images/warning.svg'
 import { getErc20InputProps } from 'lib/utils/getErc20InputProps'
+import { usePoolChainValues } from 'lib/hooks/usePoolChainValues'
+import { usePrizePoolContracts } from 'lib/hooks/usePrizePoolContracts'
+import { usePrizePoolType } from 'lib/hooks/usePrizePoolType'
+import { useOnTransactionCompleted } from 'lib/hooks/useOnTransactionCompleted'
 
 export const DepositForm = (props) => {
   const { inputError, handleSubmit, vars, stateSetters } = props
 
   const { walletMatchesNetwork } = useNetwork()
-  const [poolChainValues] = useAtom(poolChainValuesAtom)
-  const [usersChainValues] = useAtom(userChainValuesAtom)
-  const hasApprovedBalance = usersChainValues.underlyingTokenIsApproved
-  const supportsAllowance = usersChainValues.underlyingTokenSupportsAllowance
-
-  const { usersTokenBalance } = usersChainValues || {}
+  const { data: poolChainValues, isFetched: poolChainValuesIsFetched } = usePoolChainValues()
+  const { data: usersChainValues, isFetched: usersChainValuesIsFetched } = useUserChainValues()
 
-  const { tokenDecimals, isRngRequested } = poolChainValues || {}
+  if (!poolChainValuesIsFetched || !usersChainValuesIsFetched) return null
```

Comment:
Should we show a loading component here instead of empty space? I imagine if it's `null` it will start at 0px and jump to it's content-filled height in 1 render paint, which might be a bit janky?

Reply:
This line actually doesn't get hit. I'll delete it.

If there's no wallet connected: `DepositForm` doesn't get rendered
If there is: Loading screen is shown until ALL data is fetched

There is a slight flicker when going from no wallet -> wallet if your internet is too fast. Otherwise you see the loading indicator clearly. There's no bouncing around of components.